integration_test.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	cli "gopkg.in/urfave/cli.v1"
+)
+
+func TestFetchWithBranchOption(t *testing.T) {
+	t.Parallel()
+
+	tmpDownloadPath := createTempDir(t, "fetch-branch-test")
+
+	cases := []struct {
+		name       string
+		repoUrl    string
+		branchName string
+		sourcePath string
+	}{
+		// Test on a public repo whose sole purpose is to be a test fixture for this tool
+		{"branch option with public repo", "https://github.com/gruntwork-io/fetch-test-public", "sample-branch", "/"},
+
+		// Private repo equivalent
+		{"branch option with private repo", "https://github.com/gruntwork-io/fetch-test-private", "sample-branch", "/"},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
Add `t.Parallel()`?

Reply:
@brikis98 hmmm I'm a bit skeptical of a `t.Parallel()` because I'm doing a dodgy hack to override `os.Stdout` and `os.Stderr`. Should I simply comment I'm deliberately not doing that instead?

Reply:
Let's resolve how the test will work below. If we stick with weird `stdout` stuff, then yea, commenting it out explicitly and explaining why is  good idea.

Reply:
added a comment on why I'm not using parallelism.